packages/nuxt/src/pages/module.ts

```diff
@@ -86,9 +94,9 @@ export default defineNuxtModule({
 
       return false
     }
-    nuxt.options.pages = await isPagesEnabled()
```

Comment:
Not setting `nuxt.options.pages` leaves it `undefined` 🤔 is that intended?

Reply:
you shouldn't need to set it.

undefined should mean it auto detects

will check - is this throwing an error?

Reply:
In the i18n module we're using it to check whether routing/pages are set up during the build step, but I suppose that's not really the intended use of `nuxt.options.pages`.

Will check if we need it or if there's a different way to detect it, not sure if there are other modules using it in the same way.

Reply:
Now that it isn't a plain bool you can check `nuxt.options.pages` ("legacy" if manually set) and `nuxt.options.pages.enabled` I suppose 🤔

Reply:
> Now that it isn't a plain bool you can check `nuxt.options.pages` ("legacy" if manually set) and `nuxt.options.pages.enabled` I suppose 🤔

The change in this PR leaves the `nuxt.options.pages` untouched, previously it would be overwritten with the value returned by `isPagesEnabled()`. So modules will only know if/when users have manually set the pages options, not whether a pages dir has been detected and enabled router setup.

Reply:
this pr should set options.pages.enabled

Reply:
<img width="1246" alt="image" src="https://github.com/user-attachments/assets/b08227c8-9581-4dcc-9b06-45cf57960517" />

It should be detectable through the `enabled` property 🤔

Reply:
Check out https://github.com/nuxt/nuxt/pull/31101, I tried to make it more consistent with the previous behavior and added tests to check the `nuxt.options.pages` value.